tests/Fixtures/inc/functions/getRocketI18nToPreserve.php

```diff
@@ -0,0 +1,466 @@
+			'get_rocket_i18n_code'     => null,
+			'get_rocket_i18n_home_url' => null,
+			'get_rocket_parse_url'     => null,
+		],
+	],
+	'false_ShouldReturnEmptyArrayWhenLangNotString'                   => [
+		'current_lang' => 10, // not string.
+		'config'       => [
+			'data'            => [],
+			'i18n_plugin'     => false,
+			'rocket_has_i18n' => false,
+		],
+		'expected'     => [],
+		'mocks'        => [
+			'rocket_has_i18n'          => null, // not called.
+			'get_rocket_i18n_code'     => null,
+			'get_rocket_i18n_home_url' => null,
+			'get_rocket_parse_url'     => null,
+		],
+	],
+	'false_ShouldReturnEmptyArrayWhenNoPlugin'                        => [
+		'current_lang' => 'en',
+		'config'       => [
+			'data'            => [],
+			'i18n_plugin'     => false,
+			'rocket_has_i18n' => false,
+		],
+		'expected'     => [],
+		'mocks'        => [
+			'get_rocket_i18n_code'     => null,
+			'get_rocket_i18n_home_url' => null,
+			'get_rocket_parse_url'     => null,
+		],
+	],
+	'false_ShouldReturnEmptyArrayWhenNoPlugin'                        => [
+		'current_lang' => 'en',
+		'config'       => [
+			'data'            => [
+				'codes' => [ 'de', 'en', 'fr' ],
+				'langs' => [ 'de', 'en', 'fr' ],
+			],
+			'i18n_plugin'     => 'baz',
+			'rocket_has_i18n' => false,
+		],
+		'expected'     => [],
+		'mocks'        => [
+			'get_rocket_i18n_code'     => null, // not called.
+			'get_rocket_i18n_home_url' => null,
+			'get_rocket_parse_url'     => null,
+		],
+	],
+
+	/**
+	 * WPML
+	 */
+	'wpml_ShouldReturnFrAndDe'                                        => [
+		'current_lang' => 'en',
+		'config'       => [
+			'data'            => $i18n_plugins['wpml'],
+			'i18n_plugin'     => 'wpml',
+			'rocket_has_i18n' => 'wpml',
+		],
+		'expected'     => [
+			'vfs://public/wp-content/cache/wp-rocket/example.org(.*)/',
+			'vfs://public/wp-content/cache/wp-rocket/example.org(.*)/',
+		],
+		'mocks'        => [
+			'get_rocket_i18n_code'     => $i18n_plugins['wpml']['langs'],
+			'get_rocket_i18n_home_url' => [
+				'fr' => 'http://example.org?lang=fr',
+				'de' => 'http://example.org?lang=de',
+			],
+			'get_rocket_parse_url'     => [
+				'http://example.org?lang=fr' => [
+					'host'     => 'example.org',
+					'path'     => '',
+					'scheme'   => 'http',
+					'query'    => 'lang=fr',
+					'fragment' => '',
+				],
+				'http://example.org?lang=de' => [
+					'host'     => 'example.org',
+					'path'     => '',
+					'scheme'   => 'http',
+					'query'    => 'lang=de',
+					'fragment' => '',
+				],
+			],
+		],
+	],
+	'wpml_ShouldReturnEnAndDe'                                        => [
+		'current_lang' => 'fr',
+		'config'       => [
+			'data'            => $i18n_plugins['wpml'],
+			'i18n_plugin'     => 'wpml',
+			'rocket_has_i18n' => 'wpml',
+		],
+		'expected'     => [
+			'vfs://public/wp-content/cache/wp-rocket/example.org(.*)/',
+			'vfs://public/wp-content/cache/wp-rocket/example.org(.*)/',
```

Comment:
@Tabrisrp This is the behavior we were talking about. When the i18n plugin, which in this case is WPML, returns the language as a query string, i.e. `?lang=fr`, currently this function doesn't handle it. We can fix this in a separate PR.